server/channels/app/notification.go

```diff
@@ -21,13 +21,39 @@ import (
 	"github.com/mattermost/mattermost/server/v8/channels/store"
 )
 
+const (
+	StatusBlocked     = "blocked"
+	StatusServerError = "server_error"
+	StatusNotSent     = "not_sent"
+
+	TypeEmail     = "email"
+	TypeWebsocket = "websocket"
+	TypePush      = "push"
+
+	ReasonServerConfig   = "server_config"
+	ReasonUserConfig     = "user_config"
+	ReasonUserStatus     = "user_status"
+	ReasonFetchError     = "error_fetching"
+	ReasonServerError    = "server_error"
+	ReasonMissingProfile = "missing_profile"
+	ReasonPushProxyError = "push_proxy_error"
```

Comment:
All of this looks indicates what you really want is Grafana graphs, to understand the distribution of the type of error and how frequent they are. Wondering if we can simply add these metrics to Prom?

Reply:
The plan is to add these metrics to Prometheus as well, and I was planning on abstracting a lot of the direct logging calls into a separate module responsible for both writing to the log and tracking the metrics for Prometheus.